src/profile/v100/app_preferences/asset_preferences/asset_preferences.rs

```diff
@@ -0,0 +1,162 @@
+use crate::prelude::*;
+use core::hash::Hash;
+use std::hash::Hasher;
+
+/// User off-ledger preferences regarding assets.
+#[derive(
+    Debug,
+    Default,
+    Serialize,
+    Deserialize,
+    PartialEq,
+    Eq,
+    Clone,
+    derive_more::Display,
+    uniffi::Record,
+)]
+#[serde(rename_all = "camelCase")]
+#[display("assets: {:#?}", self.assets)]
+pub struct AssetPreferences {
+    #[serde(default)]
+    pub assets: HashMap<AssetAddress, AssetPreference>,
+}
+
+impl Hash for AssetPreferences {
+    fn hash<H: Hasher>(&self, state: &mut H) {
```

Comment:
and using the `IdentifiedVec` will allow you to derive `Hash` instead of manually  implementing it.